JS/Lesson20/goksel/exercises.js

```diff
@@ -64,7 +69,59 @@ console.log(square);
 
 
 
-
+console.log("----Ex3-----");
 // 3. Create a Vehicle class with make, model, and year properties, and a start() method that logs a message to the console indicating that the vehicle has started. Then create a Car class that extends the Vehicle class and adds a numDoors property. Override the start() method in the Car class to log a different message to the console indicating that the car has started.
 
+
+
+class Vehicle {
+    constructor(make, model, year){
+        this._make = make;
+        this._model = model;
+        this._year = year;
+    }
+
+    start() {
+        return `Name with ${this._make} ${this._model} car has started at ${this._year}.`
+    }
+}
+
+class Car extends Vehicle {
+    constructor(make, model, year, numDoors){
+        super(make,model,year);
+        this._numDoors = numDoors;
+    }
+    start() {
+        return `Name with ${this._make} ${this._model} car with ${this._numDoors} doors has started at ${this._year}.`
+    }
+}
+
+const suv = new Vehicle("Mercedes", "Gla", "2020");
+const minibus = new Car("Ford", "Connect", "2013", 4);
+console.log(suv.start());
+console.log(minibus.start());
+
+
+console.log("----Ex4-----");
 // 4. Create a BankAccount class with balance and interestRate properties, and deposit() and withdraw() methods. Add a transactionHistory property that is an array of objects representing each transaction made on the account, with properties for type (either "deposit" or "withdrawal"), amount, and date ({type: "withdrawal", amount: amount, date: new Date().toLocaleDateString()}). Implement deposit() and withdraw() methods that add new transactions to the transactionHistory array. Implement a get transactionHistory() getter method that returns a copy of the transaction history array, and a get currentBalance() getter method that calculates and returns the current balance of the account based on the transaction history.
+
+/*
+class BankAccount {
+    constructor(balance, interestRate){
+        this._balance = balance;
+        this._interestRate = interestRate;
+    }
+    constructor(transactionHistory ){
+        this._transactionHistory = [];
+    }
+    deposit() {
+
+    }
+
+    withdraw() {
+
+    }
+}
+*/
+
+// I couldnt even understand whag ı should do :( ...
```

Comment:
I've tried to translate requirement to Turkish, could you try again. Basically, deposit and withdrwa method should add translaction objects to _transactionHistory array and update _balance.

Then, it should be possible to get the _balance and _transactionHistory by calling getter methods.